syntax = "proto3";

option csharp_namespace = "GrpcAgriculture";

package greet;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply);
  rpc GetSensorData (SensorRequest) returns (SensorReply);
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1;
}

// Request message containing the sensor ID
message SensorRequest {
  int32 data_id = 1;
}

// Response message containing sensor data
message SensorReply {
  int32 data_id = 1;
  int32 temperature = 2;
  int32 humidity = 3;
  int32 water_level = 4;
  string timestamp = 5;
}
